### PR TITLE
API Validation Updates

### PR DESCRIPTION
**Pull request by colinfrederickson**
- API validation framework
- Employee validations updated to include employees as a parameter (will need to be updated with the HCM.Show endpoint)
- index.ts updated to allow for a single API call to be made to grab the list of employees
- apiValidation.ts introduced with logic to use for validation

Aims to address: https://www.notion.so/flatfile/API-Based-Validations-in-Record-Hooks-f75a5e39385b4dcd9fefe86b16628fd2?pvs=4#b711da63df6b47a0866727fc71fd1bc3

### CODE REVIEW

File: common/utils/request.js [hunk 1]
```diff
@@ -54,3 +54,61 @@ export const post = async ({ hostname = 'hcm.show', path, body }) => {
     console.error(`Fetch error: ${JSON.stringify(err, null, 2)}`);
   }
 };
+export const getEmployees = async ({ hostname = 'hcm.show', path, params }) => {
```
Comment (johnmosesman) [inline common/utils/request.js]: A better name for this function would be `get`, there's nothing anything employee specific here

File: recordHooks/employees/apiValidation.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      // Ensure both the applicantId and employee.id are of the same type before comparing
+      // If applicantId is a string, compare it to a string version of employee.id
+      if (typeof employeeId === 'string') {
+        return employee.employeeId === employeeId;
+      } else {
+        // If applicantId is not a string (assuming it's a number), compare it to a number version of employee.id
+        const employeeIdNum = parseInt(employee.id, 10);
+        return employeeIdNum === employeeId;
+      }
+    });
```
Comment (johnmosesman) [inline recordHooks/employees/apiValidation.ts:11]: Did something change in the dataset that causes this to be a string or an integer?

File: recordHooks/employees/employeeValidations.ts [hunk 3]
```diff
@@ -42,13 +43,6 @@ export function employeeValidations(record) {
     // Handle or rethrow the error as needed
   }
 
-  try {
-    formatRecordDates(record, 'employees-sheet');
-  } catch (error) {
-    console.log('Error occurred during date formatting:', error);
-    // Handle or rethrow the error as needed
-  }
-
```
Comment (johnmosesman) [inline recordHooks/employees/employeeValidations.ts:51]: Do we not need to validate these dates anymore?

File: workflows/project/index.ts [hunk 4]
```diff
@@ -8,6 +8,9 @@ import { dedupeEmployees } from '../../actions/dedupe';
 import { blueprintSheets } from '../../blueprints/hcmBlueprint';
 import { validateReportingStructure } from '../../actions/validateReportingStructure';
 import { FlatfileEvent } from '@flatfile/listener';
+import axios from 'axios';
```
Comment (johnmosesman) [inline workflows/project/index.ts]: axios being used here?

File: recordHooks/employees/apiValidation.ts (hunk 2 above)
Comment (colinfrederickson) [reply recordHooks/employees/apiValidation.ts:11]: This is left over from testing data from the dummy API to account for the scenario where:

- the field in FF is a string
- the field in the API was an integer

what would be the best way to handle this based on the response from HCM.Show?

File: common/utils/request.js (hunk 1 above)
Comment (jasonlgong) [reply common/utils/request.js]: Edited

File: recordHooks/employees/employeeValidations.ts (hunk 3 above)
Comment (colinfrederickson) [reply recordHooks/employees/employeeValidations.ts:51]: This is removing a duplicate validation that snuck in!

File: recordHooks/employees/apiValidation.ts (hunk 2 above)
Comment (jasonlgong) [reply recordHooks/employees/apiValidation.ts:11]: @colinfrederickson A Q for you?

File: recordHooks/employees/employeeValidations.ts (hunk 3 above)
Comment (jasonlgong) [reply recordHooks/employees/employeeValidations.ts:51]: @colinfrederickson

File: workflows/project/index.ts (hunk 4 above)
Comment (jasonlgong) [reply workflows/project/index.ts]: Removed

File: recordHooks/employees/apiValidation.ts (hunk 2 above)
Comment (johnmosesman) [reply recordHooks/employees/apiValidation.ts:11]: hcm.show should only be returning strings

Comment (jasonlgong) [reply recordHooks/employees/apiValidation.ts:11]: Removed conditionals

**Review by johnmosesman — APPROVED** [review]

**Pull request merged**